### ISSUE TITLE
MatPlotLib plotting issues with v. large sample numbers

### ISSUE THREAD

**Issue by ewels** [MultiQC/MultiQC#408]
As reported by @varemo [on Gitter](https://gitter.im/ewels/MultiQC?at=58a1b9aa238b1dae5702f645):

> I was wondering if you have any pointers on how to handle datasets with a huge number of samples (e.g. single cell data tend to have thousands of samples). In particular I am looking at a MultiQC report from such a dataset, where the barplots for the STAR Alignment Scores and STAR Gene Counts look odd: some sample labels are "randomly" skipped, and the bars appear in chunks with white space in between (e.g. as if there was no data for those samples and thus just showing white...). Also, bars and sample labels do not align. Nevertheless, the underlying data files in `multiqc_data/` have the correct number of samples (i.e. seems to be a plotting issue). I'm looking at the flat image plots btw.

Example section of problematic plot:
![image](https://cloud.githubusercontent.com/assets/465550/23300733/38864690-fa87-11e6-80a4-7b6c8440aedd.png)

**Comment by ewels** [comment]
This plotting problem is specific to MatPlotLib and only appears with very large sample numbers. It is because I [set a maximum](https://github.com/ewels/MultiQC/blob/d168cc4934860870de3102aaf0540159110173ab/multiqc/plots/bargraph.py#L331) height for MatPlotLib bar graphs of 30 inches _(such useful units)_ to prevent plots from getting ridiculously massive.

This means that when there are a massive number of bars, MatPlotLib tries to squash them into the available space. The bars get thinner and thinner until at some point, the 1px white border completely covers the bar and they vanish. If you look closely at the example plot above, you can see the grey background lines are broken with white dashes - so the bars are there, it's just that they're white on a white background.

So, to fix this I should be able to do a couple of things:

1. Remove the white border when plotting over _some number_ of samples
2. Push up the maximum height with very large numbers of samples, to ensure that each bar is at least 1px tall.

As for the "_Also, bars and sample labels do not align."_ bit, I'm not sure that there's anything I can do. My options are either to print all labels (in which case the axis will just be a black square of overlapping text), or to print every nth label as I do currently, in which case the alignment is vague at best. I think that this is probably the lesser of two evils though.

**Comment by varemo** [comment]
This sounds like a good plan! Regarding the labels, maybe skip them completely above some given number of samples?

**Comment by ewels** [comment]
So, turns out that this is harder than I expected. I disabled the white border which helps, but bars still disappear when there isn't enough plotting space. I've spent most of the afternoon chasing down the MatPlotLib rabbit hole but haven't really gotten anywhere, so I've posted a [question on StackOverflow](http://stackoverflow.com/questions/42536592/matplotlib-dynamic-plot-height-horizontal-bars-always-1px-height) - hopefully someone will help!

I actually think that the plots look better without the white border around the bars, so I've removed that default for all plots now. Let's see if anyone notices 😉
